main/src/cgeo/geocaching/ui/dialog/SimpleDialog.java

```diff
@@ -0,0 +1,446 @@
+package cgeo.geocaching.ui.dialog;
+
+import cgeo.geocaching.R;
+import cgeo.geocaching.activity.Keyboard;
+import cgeo.geocaching.ui.TextParam;
+import cgeo.geocaching.ui.ViewUtils;
+import cgeo.geocaching.utils.functions.Action1;
```

Comment:
What's the difference between our self created `Action1` and `Consumer`?

Reply:
There is no difference at all. I assume that `Action1` was created either before `androidx.core.util.Consumer` existed or for naming and usage consistency towards `Action2` (for which there is no equivalent in `androidx.core.util`)

The deeper underlying problem here is that the real function classes of JDK under `java.util.function` can only be used starting with SDK Level 24 while c:geo runs on SDK level 21. There would also be a `java.util.function.Consumer` class, as well as e.g. `BiConsumer `(replacement for `Action2`), `Function`/`BiFunction` (replacement for `Func1`/`Func2`), `Predicate`and many many more. I think for now it doesn't really matter whether we use Action1 or `androidx.core.util.Consumer` because once we switch to SDK 24, we should anyway convert all usages of both `androidx.core.util` and `cgeo.geocaching.utils.functions` to standard JDK functions in `java.util.function` anyway.

But if you insist I can change `Action1 `to `androidx.core.util.Consumer`. Should I?

Reply:
Ah, what the heck, I changed `Action1` to `Consumer`...

Reply:
> But if you insist I can change Action1 to androidx.core.util.Consumer. Should I?

Up to you, I was just wondering if there's any difference. Maybe we should change all usages to one of both methods to be consistent in our code...? But that's definitely unrelated ;-)